### PR TITLE
[Refactoring] fix Reactor Intellij inspections

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java [hunk 1]
```diff
@@ -274,15 +274,25 @@ public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws Mailb
...
+        Mono<ModSeq> nextModSeqMono = modSeqProvider.nextModSeq(mailboxId)
+            .switchIfEmpty(Mono.error(() -> new MailboxException("Can not find a MODSEQ to save " + message.getMessageId() + " in " + mailboxId)));
+
+        try {
+            Mono.zip(messageUidMono, nextModSeqMono)
+                .doOnNext(tuple -> {
+                    message.setUid(tuple.getT1());
+                    message.setModSeq(tuple.getT2());
+                })
+                .block();
```
Comment (remk) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java]: maybe it as not its place in this PR, but this method could return a Mono, and the `add` method flatmap over its result

**Comment by Arsnael** [comment]
```
[611ac31fc0ec91bd21df3b87a54109fc67752d68] [ERROR] Failures: 
[611ac31fc0ec91bd21df3b87a54109fc67752d68] [ERROR]   CassandraBlobStoreTest.deleteBucketShouldDeleteExistingBucketWithItsData 
[611ac31fc0ec91bd21df3b87a54109fc67752d68] Expecting code to raise a throwable.
[611ac31fc0ec91bd21df3b87a54109fc67752d68] [ERROR]   CassandraBlobStoreTest.deleteShouldDeleteExistingBlobData 
[611ac31fc0ec91bd21df3b87a54109fc67752d68] Expecting code to raise a throwable.
[611ac31fc0ec91bd21df3b87a54109fc67752d68] [ERROR]   CassandraBlobStoreTest.readShouldThrowWhenNoExistingStream 
[611ac31fc0ec91bd21df3b87a54109fc67752d68] Expecting code to raise a throwable.
[611ac31fc0ec91bd21df3b87a54109fc67752d68] [ERROR]   CassandraBlobStoreTest.readStringShouldThrowWhenBucketDoesNotExist 
[611ac31fc0ec91bd21df3b87a54109fc67752d68] Expecting code to raise a throwable.
```

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java (hunk 1 above)
Comment (mbaechler) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java]: Yes, I tried to focus on the inspections results for this PR

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**